checker/tests/nullness/Issue2432.java

```diff
@@ -0,0 +1,112 @@
+// Test case for issue 2432:
+// https://github.com/typetools/checker-framework/issues/2432
+
+import java.util.List;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.checker.nullness.qual.PolyNull;
+
+class Issue2432 {
```

Comment:
Can you clean up this test case a bit?
Add a bit of structure and clean naming. Explain a bit (very briefly) what is tested in each section.